tests/opt-testcases/tls13-compat.sh

```diff
@@ -3274,3 +3274,643 @@ run_test "TLS 1.3 m->G: TLS_AES_128_CCM_8_SHA256,x448,rsa_pss_rsae_sha256" \
          -c "server hello, chosen ciphersuite: ( 1305 ) - TLS1-3-AES-128-CCM-8-SHA256" \
          -c "Certificate Verify: Signature algorithm ( 0804 )" \
          -c "Verifying peer X.509 certificate... ok"
+
```

Comment:
Should we only change the pattern in hrr test cases? Should the exist test cases be changed as the pattern either?

Reply:
For this part: "curves=x448,secp384r1,x25519", seems it's difficult to know the curves selected by server peer when I initialize the client. So I have to  list all of the left curves with low priority in client.

Reply:
> Should we only change the pattern in hrr test cases? Should the exist test cases be changed as the pattern either?

Yes only HRR test cases. It is good to be sure we go through all possible handshake parameters and the cost in terms of number of test case is reasonable. With HRR there is just to many test cases with this systematic approach thus we have to do differently.

Reply:
> For this part: "curves=x448,secp384r1,x25519", cause we initialize the client and server seperately, seems it's difficult to know the curves selected by server peer when I initialize the client. So I have to list all of the left curves with low priority in client.

My understanding is that both client and server objects are created in `generate_compat_hrr_test` where we know both the client initial curve and the server curve thus it should be ok to just have those two curves. Otherwise regarding `secp384r1` I would suggest to hardcode it in `generate_compat_hrr_test` with a message saying that it should be removed when the issue described in #5646 has been resolved.

Reply:
I have update #5659 and add hrr tests. Please check.

This PR can preceding it and revert the last commit "skip all hrr test"

Reply:
Maybe you can remove the HRR test cases in #5659, and I will add them here. @yuhaoth

Reply:
Yes please do so. I've suggested that in #5659 as well.